pkg/tecdsa/dkg/dkg.go

```diff
@@ -85,6 +86,101 @@ func (e *Executor) Execute(
 	return finalizationState.result(), endBlockNumber, nil
 }
 
+// StartedDKGEvent represents a DKG start event.
+type StartedDKGEvent struct {
+	Seed        *big.Int
+	BlockNumber uint64
+}
+
+// ResultSubmissionEvent represents a DKG result submission event. It is emitted
+// after a submitted DKG result is positively validated on the chain. It contains
+// the index of the member who submitted the result and a final public key of
+// the group.
+type ResultSubmissionEvent struct {
+	MemberIndex         uint32
+	GroupPublicKeyBytes []byte
+	Misbehaved          []uint8
+
+	BlockNumber uint64
+}
+
+type ResultSubmitter interface {
+	// SubmitDKGResult sends DKG result to a chain, along with signatures over
+	// result hash from group participants supporting the result.
+	// Signatures over DKG result hash are collected in a map keyed by signer's
+	// member index.
+	SubmitDKGResult(
+		participantIndex group.MemberIndex,
+		result *Result,
+		signatures map[group.MemberIndex][]byte,
+	) error
+	// IsGroupRegistered checks if group with the given public key is registered
+	// on-chain.
+	IsGroupRegistered(groupPublicKey []byte) (bool, error)
+	// OnDKGResultSubmitted registers a callback that is invoked when an on-chain
+	// notification of a new, valid submitted result is seen.
+	OnDKGResultSubmitted(
+		func(event *ResultSubmissionEvent),
+	) subscription.EventSubscription
+}
+
+type ResultSigner interface {
+	// Signing returns the chain's signer.
+	Signing() chain.Signing
+	// CalculateDKGResultHash calculates 256-bit hash of DKG result in standard
+	// specific for the chain. Operation is performed off-chain.
+	CalculateDKGResultHash(result *Result) (ResultHash, error)
```

Comment:
What do you think about exposing methods like:
```
SignResult(result *Result) (ResultSignature, ResultHash, error)
VerifyResult(result *Result, signature ResultSignature, publicKey []byte) (bool, error)
```
where the `ResultSignature` is just a `byte[]`. Then, we can implement that interface in `pkg/tbtc`

Reply:
Done in c46033aafc6a094ee8380749b1b0ac164db3cb21.